File: gui/src/integrations/storage/artifacts.ts
import type { StoredArtifact, WorkspaceFileResult } from "./types";
import { invokeCommand } from "../tauri/invoke";

// ─── Artifacts ───────────────────────────────────────────────────────────

export async function listArtifacts(threadId: string) {
  return invokeCommand<StoredArtifact[]>("list_artifacts", { threadId });
}

export async function createArtifact(input: {
  workspaceId: string;
  threadId?: string | null;
  runId?: string | null;
  title: string;
  artifactType: string;
  path?: string | null;
  content?: string | null;
  contentStorage?: string | null;
  summary?: string | null;
}) {
  return invokeCommand<StoredArtifact>("create_artifact", { input });
}

export async function importAttachmentArtifact(input: { threadId: string; path: string }) {
  return invokeCommand<StoredArtifact>("import_attachment_artifact", { input });
}

export async function deleteArtifact(artifactId: string) {
  return invokeCommand<StoredArtifact>("delete_artifact", { artifactId });
}

// ─── Workspace file `@`-mention search ─────────────────────────────────────

export async function searchWorkspaceFiles(input: {
  workspaceId: string;
  query?: string | null;
  limit?: number | null;
}) {
  return invokeCommand<WorkspaceFileResult[]>("search_workspace_files", { input });
}

File: gui/src-tauri/src/commands/artifacts.rs
//! Artifact Tauri commands.

use crate::store;

#[tauri::command]
pub fn list_artifacts(thread_id: String) -> Result<Vec<store::ArtifactRecord>, crate::AppError> {
    store::list_artifacts(&thread_id)
}

#[tauri::command]
pub fn create_artifact(
    input: store::CreateArtifactInput,
) -> Result<store::ArtifactRecord, crate::AppError> {
    store::create_artifact(input)
}

#[tauri::command]
pub fn import_attachment_artifact(
    input: store::ImportAttachmentArtifactInput,
) -> Result<store::ArtifactRecord, crate::AppError> {
    store::import_attachment_artifact(input)
}

#[tauri::command]
pub fn delete_artifact(artifact_id: String) -> Result<store::ArtifactRecord, crate::AppError> {
    store::delete_artifact(&artifact_id)
}
